### DISCUSSION TITLE
Why did you create your own migration tool?

### DISCUSSION THREAD

**Discussion by poorwang** [pocketbase/pocketbase#2480] [Q&A] [answered]
Hello, I'm curious why you decided to implement your own migration tool instead of using something like https://github.com/golang-migrate/migrate?

Is it because you want to ship a single binary and these tools would prevent that with their generated .sql files, or is there another reason?

**Comment by ganigeorgiev** [comment, marked as answer]
Mainly because the other tools were far too complicated for the needs of PocketBase. We need only "up" (apply new) and "down" (revert).

I also wanted the migrations to be simple Go files so that users can execute regular Go code allowing to run something only once, to adjust existing data to fit new schema, etc.
